src/loaders/lottie/tvgLottieLoader.cpp

```diff
@@ -288,45 +291,133 @@ Paint* LottieLoader::paint()
+
+    if (slotId == 0) {
+        ARRAY_FOREACH(p, comp->slots) {
+            (*p)->reset();
+        }
+        overridden = false;
+        rebuild = true;
+        return true;
+    }
+
+    LottieSlotData* slotData = nullptr;
+
+    INLIST_FOREACH(comp->slotDatas, p) {
+        if (p->id == slotId) {
+            slotData = p;
+            break;
+        }
+    }
+
+    if (!slotData) return false;
+
+    ARRAY_FOREACH(p, slotData->datas) {
+        ARRAY_FOREACH(q, comp->slots) {
+            if (strcmp((*q)->sid, p->sid)) continue;
+            (*q)->apply(p->prop, byDefault);
+            break;
+        }
+    }
+
+    overridden = true;
+    rebuild = true;
+    return true;
+}
 
-    //override slots
-    if (slots) {
-        //Copy the input data because the JSON parser will encode the data immediately.
-        auto temp = byDefault ? slots : duplicate(slots);
 
-        //parsing slot json
-        LottieParser parser(temp, dirName, builder->expressions());
-        parser.comp = comp;
+bool LottieLoader::delSlot(uint32_t slotId, bool byDefault)
+{
+    if (!ready() || comp->slots.count == 0 || comp->slotDatas.empty()) return false;
 
-        auto idx = 0;
-        auto succeed = false;
-        while (auto sid = parser.sid(idx == 0)) {
-            auto applied = false;
+    if (slotId == 0) {
+        if (!byDefault) {
             ARRAY_FOREACH(p, comp->slots) {
-                if (strcmp((*p)->sid, sid)) continue;
-                if (parser.apply(*p, byDefault)) succeed = applied = true;
-                break;
+                (*p)->reset();
             }
-            if (!applied) parser.skip();
-            ++idx;
+            rebuild = true;
         }
-        tvg::free((char*)temp);
-        rebuild = succeed;
-        overridden |= succeed;
-        return rebuild;
-    //reset slots
-    } else if (overridden) {
-        ARRAY_FOREACH(p, comp->slots) (*p)->reset();
+
+        comp->slotDatas.free();
         overridden = false;
-        rebuild = true;
+        return true;
     }
+
+    LottieSlotData* slotData = nullptr;
+
+    INLIST_FOREACH(comp->slotDatas, p) {
+        if (p->id == slotId) {
+            slotData = p;
+            break;
+        }
+    }
+
+    if (!slotData) return false;
+
+    ARRAY_FOREACH(p, slotData->datas) {
+        ARRAY_FOREACH(q, comp->slots) {
+            if (strcmp((*q)->sid, p->sid)) continue;
+            if (!byDefault && (*q)->overridden) {
+                (*q)->reset();
+                rebuild = true;
+            }
+            break;
+        }
+    }
+
+    comp->slotDatas.remove(slotData);
+    delete(slotData);
     return true;
 }
 
 
+uint32_t LottieLoader::genSlot(const char* slots)
+{
+    if (!slots || !ready() || comp->slots.count == 0) return 0;
+
+    auto slotId = djb2Encode(slots);
+    INLIST_FOREACH(comp->slotDatas, p) {
+        if (p->id == slotId) return slotId;
+    }
+
+    auto temp = duplicate(slots);
```

Comment:
Previously, that slots is skipped copying in the `byDefault` case because the text is hold by the parser. Actually, after the slot data is generated and the slot text can be flushed out immediately. Please double check this.